src/jit/codegenxarch.cpp

```diff
@@ -4715,8 +4725,24 @@ instruction CodeGen::genGetInsForOper(genTreeOps oper, var_types type)
 void CodeGen::genCodeForShift(GenTreePtr tree)
 {
     // Only the non-RMW case here.
+#ifdef _TARGET_64BIT_
     assert(tree->OperIsShiftOrRotate());
-    assert(!tree->gtOp.gtOp1->isContained());
+#else
+    assert(tree->OperIsShiftOrRotate() || tree->OperGet() == GT_LSH_HI || tree->OperGet() == GT_RSH_LO);
```

Comment:
Why can't those two new opcodes be added to OperIsShiftOrRotate so that you dont have to special case them here.